### DISCUSSION TITLE
More Programming Exercises for students who finished the script

### DISCUSSION THREAD

**Discussion by Xeldlie** [FachschaftMathPhysInfo/Programmiervorkurs#146] [Ideas]
What are good exercises for students who just learned the basics of functional programming?
Exercises the students can expand and iterate upon are welcome

**Comment by fschledorn** [comment]
Ideas so far:

- Fibonacci: Laufzeitvergleich Iterativ, Rekursiv, Rekursiv mit Speicher
- Schiffe versenken (vereinfacht)
- Blackjack Bot
- Schere, Stein, Papier
- Würfelsimulator
- Anagramme: Der Computer gibt ein Wort als Anagramm (Buchstaben durcheinander) aus und der Spieler muss das ursprüngliche Wort erraten
- Sudoku
- Sudoku solver

**Reply by Tim-Grube** [reply]
Hangman

**Comment by marmolotte** [comment]
-Fizzbuzz erweiterbar machen

**Reply by fschledorn** [reply]
Was genau meinst du damit? Möchtest du, dass man am Anfang wählen kann welche Zahlen auf Teilbarkeit durch 3 und 5 überprüft werden sollen oder willst du, dass man angeben kann welche Teiler benutzt werden sollen. Oder möchtest du beides kombinieren?
